gopls/internal/lsp/template/parse.go

```diff
@@ -108,22 +110,13 @@ func parseBuffer(buf []byte) *Parsed {
 	}
 	ans.setTokens() // ans.buf may be a new []byte
 	ans.lines = bytes.Split(ans.buf, []byte{'\n'})
-	t, err := template.New("").Parse(string(ans.buf))
-	if err != nil {
-		funcs := make(template.FuncMap)
-		for t == nil && ans.ParseErr == nil {
-			// in 1.17 it may be possible to avoid getting this error
-			//  template: :2: function "foo" not defined
-			matches := parseErrR.FindStringSubmatch(err.Error())
-			if len(matches) == 2 {
-				// suppress the error by giving it a function with the right name
-				funcs[matches[1]] = func() interface{} { return nil }
-				t, err = template.New("").Funcs(funcs).Parse(string(ans.buf))
-				continue
-			}
-			ans.ParseErr = err // unfixed error
-			return ans
-		}
+	t := template.New("")
+	tr := parse.New("")
+	tr.Mode = parse.SkipFuncCheck | parse.ParseComments
```

Comment:
Using `SkipFuncCheck` here removes quite a bit of code, which is a nice coincidence with needing `ParseComments`